openest/generate/functions.py

```diff
@@ -443,6 +443,148 @@ def describe():
                     description="Sum the results of multiple previous calculations.")
 
 
+
+class FractionSum(calculation.Calculation):
+    """Sum of two subcalculations weighted on fraction of unity.
+
+    In short, ``frac * subcalc[0] + (1 - frac) * subcalc[1]``.
+
+    Parameters
+    ----------
+    frac : ``openest.generate.calculation.Calculation``
+        Result gives weight of first subcalculation. 1 - `frac` is weight of
+        second subcalculation. Must be ≤ 1. The
+    subcalcs : Sequence of ``openest.generate.calculation.Calculation``
+        Calculations to weight. Must be ``len(subcalcs) == 2``. The units of
+        the subcalcs elements must be the same.
+    unshift : bool, optional
+    """
+
+    def __init__(self, frac, subcalcs, unshift=True):
```

Comment:
The automatic argument matcher will get confused by splitting out the first of three subcalcs, and there's a second reason to keep them all in one list: you can generalize this to N subcalcs, by saying that they go in the following order:
`fraction1, subcalc1, [fraction2, subcalc2, ...], subcalcN`
where the fraction applied to `subcalcN` is just (1 - (fraction1 + fraction2 + ...)).

If you don't plan on setting this up in the general way, then I think you should just have three separate arguments:
`fraccalc`, `truecalc`, `falsecalc`
or similarly named.

Reply:
Resolved with acc586a.